symbolic-ppdb/src/format/sourcelinks.rs

```diff
@@ -0,0 +1,186 @@
+use std::cmp::Ordering;
+
+use crate::{FormatError, FormatErrorKind};
+
+/// See [Source Link PPDB docs](https://github.com/dotnet/designs/blob/main/accepted/2020/diagnostics/source-link.md#source-link-json-schema).
+#[derive(Clone)]
+pub(crate) struct SourceLinkMappings {
+    rules: Vec<Rule>,
+}
+
+#[derive(Clone)]
+struct Rule {
+    pattern: Pattern,
+    url: String,
+}
+
+#[derive(Clone)]
+enum Pattern {
+    Exact(String),
+    Prefix(String),
+}
+
+impl SourceLinkMappings {
+    pub fn empty() -> Self {
```

Comment:
You can just `#[derive(Default)]` for this.

Reply:
👍